### PR TITLE
Add tutorials, examples, and guides for the Transformers Flavor

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/10360?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10360/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10360
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Refactor the content in models.rst to move the transformers information to an "in-depth" guide.
Migrate the existing examples to notebook-based tutorials
Provide an overview landing page for the flavor that explains the flavor at a high level

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [X] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Overhaul the transformers flavor docs and add use-case tutorials

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
This font is difficult to read. Can we use a different font?

image

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+   "source": [
+    "import transformers\n",
+    "\n",
+    "import mlflow\n",
+    "\n",
+    "task = \"text2text-generation\"\n",
+    "\n",
+    "generation_pipeline = transformers.pipeline(\n",
+    "    task=task,\n",
+    "    model=\"declare-lab/flan-alpaca-large\",\n",
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:174]: can we use a smaller model?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+    "\n",
+    "The function takes a list of prediction strings and formats each one by:\n",
+    "\n",
+    "- Splitting the prediction text into individual sentences.\n",
+    "- Ensuring that each sentence is properly stripped of leading/trailing whitespace and ends with a period.\n",
+    "- Joining the sentences back together with newline characters for clear visual separation.\n",
+    "\n",
+    "This formatting is particularly useful in a notebook setting where we want to present the output in a more readable and presentable manner for demonstration purposes."
+   ]
+  },
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:385]: Do we need this explanation?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction to MLflow and Transformers\n",
+    "\n",
+    "Welcome to our first tutorial on leveraging the power of **Transformers** with **MLflow**. This tutorial is designed for beginners who are just getting started with machine learning workflows and model management. We will walk you through a simple example that demonstrates the integration of the Transformers library with MLflow, a platform that simplifies the machine learning lifecycle.\n",
+    "\n",
+    "### What are Transformers?\n",
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:11]: do we need to explain what transformers are?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+    "    mlflow.transformers.generate_signature_output(generation_pipeline, input_example),\n",
+    "    parameters,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting the tracking server\n",
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:243]: Can we remove this section? For demonstrating purposes, I think we can just use the default local file store.

File: docs/source/llms/transformers/tutorials/translation/component-translation.ipynb [hunk 5]
```diff
@@ -0,0 +1,749 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction to Translation with Transformers and MLflow\n",
+    "\n",
+    "In this tutorial, we delve into the world of language translation by leveraging the power of [Transformers](https://huggingface.co/docs/transformers/model_doc/whisper) and MLflow. This guide is crafted for practitioners with a grasp of machine learning concepts who seek to streamline their translation model workflows. We will showcase the use of MLflow to log, manage, and serve a cutting-edge translation model - the `google/flan-t5-base` from the [🤗 Hugging Face](https://huggingface.co/) library.\n",
+    "\n",
+    "### What is `flan-t5-base`?\n",
```
Comment (harupy) [inline docs/source/llms/transformers/tutorials/translation/component-translation.ipynb:11]: Do we need to explain this? Can you just provide a link to the huggingface hub?

File: docs/source/llms/transformers/index.rst [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Fine-tuning a pre-trained model is a common practice in machine learning. MLflow streamlines this process by allowing users to log 
+the fine-tuning parameters, track the training metrics, and save the resulting model. This ensures reproducibility and facilitates 
+comparison across different experiments.
+
+To see an example of how to fine-tune a model and log the results in MLflow in an end-to-end example of creating a spam classifier pipeline, 
+see the `fine-tuning tutorial <tutorials/fine-tuning/transformers-fine-tuning.html>`_. 
+
+For a frame of reference of what the experience is within MLflow for fine-tuning a transformers foundation model, the image below shows the
```
Comment (B-Step62) [inline docs/source/llms/transformers/index.rst:80]: Could you clarify why we need two MLflow Runs for a single fine-tuning task? If I understand correctly, the green one is produced by `# Log the pipeline` part of the tutorial, but I'm not fully sure why we have to start a separate run for it, instead of within the training run which I think is more intuitive...

If having two runs is our recommended practice, we can show training run alone here and move the other run to deployment section and explain the reason behind that, cuz it's not directly relevant to experiment tracking, rather more productionization purpose. WDYT?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+    "### Why Combine MLflow with Transformers?\n",
+    "\n",
+    "Combining MLflow with Transformers brings several benefits:\n",
+    "\n",
+    "- **Experiment Tracking**: Easily log and compare parameters, metrics, and outputs from your Transformers models.\n",
+    "- **Model Management**: Keep track of different versions of your models, configured parameters, and their respective performance.\n",
+    "- **Reproducibility**: Record everything needed to reproduce a model prediction, from data preprocessing to model parameters.\n",
+    "- **Deployment**: Streamline the process of deploying your Transformers models to production.\n",
+    "\n",
+    "### What Will You Learn?\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:24]: nit: I love this kind of "objective" section! Would be even nicer if it's shown on top of the page, because I can quickly decide whether or not the tutorial is for me or not.

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 4 above)
Comment (B-Step62) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:243]: +1, it seems the need of a server (which is not true but perceptionally) is a psychological hurdle to start MLflow

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+    "\n",
+    "### Steps\n",
+    "\n",
+    "- We import the necessary libraries: transformers for building our NLP model and mlflow for model tracking and management.\n",
+    "- We then define the task for our pipeline, which in this case is `text2text-generation`. This task involves generating new text based on the input text.\n",
+    "- Next, we create a generation_pipeline using the `pipeline` function from the Transformers library. This pipeline is configured to use the `declare-lab/flan-alpaca-base model`, which is a pre-trained model suitable for text generation.\n",
+    "- For the purposes of generating a signature later on, as well as having a visual indicator of the expected input data to our model when loading as a `pyfunc`, we next set up an input_example that contains sample prompts.\n",
+    "- Finally, we define parameters that will be used to control the behavior of the model during inference, such as the maximum length of the generated text and whether to sample multiple times.\n",
+    "\n",
+    "### Understanding Pipelines\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:54]: What about collapse those introductory sections? I think we should have these description for Pipelines, Signatures, etc, for ppl who directly jumped into the tutorial, but it takes larger space than the main task.

```
<h1>Understanding Pipelines</h1>
<span>Pipelines are a high-level abstraction provided by the Transformers library that simplifies the process of using models for inference. They encapsulate the complexity of the underlying code, offering a straightforward API for a variety of tasks, such as text classification, question answering, and in our case, text generation. </span><br/><br/>

<details>
<summary>
    <b>Expand to learn about Transformer Pipelines.</b>
</summary>
<h2>The pipeline() function</h2>
<span>The pipeline() function is a versatile tool that can be used to create a pipeline for any supported task. When you specify a task, the function returns a pipeline object tailored for that task, constructing the required calls to sub-components (a tokenizer, encoder, generative model, etc.) in the order needed to fulfill the needs of the specified task. This abstraction dramatically simplifies the code required to use these models and their respective components.</span>
</details>
```
Screenshot 2023-11-20 at 11 36 38

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-fine-tuning.ipynb [hunk 9]
```diff
@@ -0,0 +1,1096 @@
+{
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/fine-tuning/transformers-fine-tuning.ipynb:1]: It seems the notebook layout becomes a bit noisy on browser, due to random spacing makes it hard to visually capture the section structure. Is there any easy way we can clean this up?

Screenshot 2023-11-20 at 11 38 30

**Review by B-Step62 — COMMENTED** [review]
Thanks a lot for really ellaborate work for the refactoring and tutorials! I love the contents and detailed navigation, just added some personal suggestions to make it visually consice:)

File: docs/source/llms/transformers/index.rst [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To download the transformers tutorial notebooks to run in your environment, click the respective links below:
+
+.. raw:: html
+
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb" class="notebook-download-btn">Download the Introductory Text Generation Notebook</a><br>
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/llms/transformers/tutorials/audio-transcription/whisper.ipynb" class="notebook-download-btn">Download the Audio Transcription Notebook</a><br>
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/llms/transformers/tutorials/translation/component-translation.ipynb" class="notebook-download-btn">Download the Translation Notebook</a><br>        
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/llms/transformers/tutorials/conversational/conversational-model.ipynb" class="notebook-download-btn">Download the Chat Conversational Notebook</a><br>
```
Comment (harupy) [inline docs/source/llms/transformers/index.rst:179]: Can we review `conversational-model.ipynb` in a separate PR?

File: docs/source/llms/transformers/index.rst [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+
+.. toctree::
+    :maxdepth: 2
+    :hidden:
+
+    tutorials/text-generation/text-generation.ipynb
+    tutorials/audio-transcription/whisper.ipynb
+    tutorials/translation/component-translation.ipynb
+    tutorials/conversational/conversational-model.ipynb
+    tutorials/fine-tuning/transformers-fine-tuning.ipynb
```
Comment (harupy) [inline docs/source/llms/transformers/index.rst:191]: Any notebooks that can be reviewed in a separate PR?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 1 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:174]: Unfortunately, the smaller version of the flan-alpaca model is only suitable for CI testing. The results that are returned from the 2 smaller models are so bad that it makes the tutorial look like it's broken. There are 4 larger models than this one which provide dramatically improved results, but they're so large that the runtime on CPU for executing the tutorial is measured in tens of minutes. This version of the model is a fairly good compromise for size vs. ability

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 2 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:385]: Good call. Removed everything but the first 2 sentences (and shortened them).

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 3 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:11]: I'm making no assumptions about users knowledge with any introduction notebook flavor. That being said, that's the only place such a high-level explanation should happen within a collection of tutorials (I made sure that the only "what is this thing that we're learning about in this tutorial" for the more advanced tutorials in transformers don't have a 'what is transformers' entry). The other tutorials focus on things like "What is Whisper?" in this spot.

File: docs/source/llms/index.rst [hunk 12]
```diff
@@ -270,6 +270,34 @@ Select the integration below to read the documentation on how to leverage MLflow
...
+Native Integration Guides and Tutorials
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. raw:: html
+
+    <section>
+        <article class="simple-grid">
+            <div class="simple-card">
+                <a href="transformers/index.html">
+                    <div class="header">
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/llms/index.rst:282]: That was a design request from UX to ensure that all cards have their <p> content start at the same height with an offset from the border of the <h2> at the top. 
I think we'll be refining the CSS for this, though. (In a later PR :) )

File: docs/source/llms/transformers/index.rst [hunk 13]
```diff
@@ -0,0 +1,279 @@
...
+enabling practitioners to manage experiments, track model versions, and serve models with unprecedented ease.
+
+The Transformers library offers:
+
+- **Pre-trained Models**: Access to a comprehensive collection of `pre-trained models <https://huggingface.co/models>`_, reducing compute costs, carbon footprint, and the time needed to train models from scratch.
+- **Task Versatility**: Support for common tasks across different modalities such as text classification, entity recognition, question answering, summarization, translation, image classification, object detection, segmentation, speech recognition, and more.
+- **Framework Interoperability**: Flexibility to train models in one framework and deploy in another, supporting PyTorch, TensorFlow, and JAX, with additional export options for ONNX and TorchScript.
+- **Community**: A highly active and engaged community for collaboration and support through forums and the Hugging Face Hub.
+
+MLflow's Transformers flavor is designed to harness the power of the Transformers library within the MLflow ecosystem, providing tools for:
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/llms/transformers/index.rst:41]: Good call. Dramatically simplifed these sections and condensed the information for a faster read.

File: docs/source/llms/transformers/index.rst [hunk 14]
```diff
@@ -0,0 +1,279 @@
...
+   :width: 80%
+   :align: center
+
+   Fine-tuning a Transformers Model with MLflow
+
+Deployment Made Easy
+^^^^^^^^^^^^^^^^^^^^
+
+Once a model is trained, it needs to be deployed for inference. MLflow's integration with Transformers simplifies this by providing 
+functions such as :py:func:`mlflow.transformers.load_model` and :py:func:`mlflow.pyfunc.load_model`, which allow for easy model serving.
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/llms/transformers/index.rst:94]: good idea! Added one for loading as a pyfunc (to the guide) and one for inference configurations for scalable serving.

File: docs/source/llms/transformers/index.rst [hunk 15]
```diff
@@ -0,0 +1,279 @@
...
+                    </div>
+                    <p>
+                        Learn how to define a custom PyFunc using transformers for advanced, state-of-the-art new models.
+                    </p>
+                </a>
+            </div>
+        </article>
+    </section>
+
+Download the Tutorial Notebooks to try them locally
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/llms/transformers/index.rst:169]: Unfortunately, we can't render these or properly trigger a local notebook download from within the Jupyter notebook render. nbsphinx (the plugin for sphinx that renders the .ipynb as .html) does not allow for custom JS or CSS to propagate to the rendered html (I tried). In order for us to actually have a notebook download to local on all browsers, we have to specify some custom JS to override the settings that the server (s3 website hosting) specifies for how to handle specific file types (otherwise, it will, by browser default, render the raw text of the notebook content directly in the browser). 
We might be able to come up with something different with a change to a different platform for html generation, but for now, we have to have some external to the notebook rendering mechanism that generates a link / button to download the files.

File: docs/source/llms/transformers/index.rst (hunk 6 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/index.rst:80]: It's not really a recommended task, per se. The original intention was to reduce the complexity. Splitting the runs up like this is fairly confusing, though. I unified them and updated the screen capture visualization as well :) Good idea

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 7 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:24]: Great call out. Updated all of the notebooks.

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 8 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:54]: This is an AWESOME IDEA! thank you so much for this. Updating the tutorials to make the notebooks cleaner :)

File: docs/source/llms/transformers/tutorials/fine-tuning/transformers-fine-tuning.ipynb (hunk 9 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/fine-tuning/transformers-fine-tuning.ipynb:1]: I think it's the styling of our template with how it formats the different <h> elements. I'm going to only use h2 at the top of the notebook and increment the header sections to try and make it more visually appealing

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 4 above)
Comment (harupy) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:243]: > Explain how to connect to hosted MLflow (Databricks CE)

Can we explain this in a different section?

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 16]
```diff
@@ -0,0 +1,505 @@
...
+    "\n",
+    "In this first section, we are setting up our environment and configuring aspects of the transformers pipeline that we'll be using to generate a text response from the LLM. \n",
+    "\n",
+    "<details>\n",
+    "    <div>\n",
+    "        <h4>Setting up our Pipeline</h4>\n",
+    "        <ul>\n",
+    "            <li><strong>Import</strong>: We import the necessary libraries: transformers for building our NLP model and mlflow for model tracking and management.</li>\n",
+    "            <li><strong>Task Definition</strong>: We then define the task for our pipeline, which in this case is <code>text2text-generation`</code> This task involves generating new text based on the input text.</li>\n",
+    "            <li><strong>Pipeline Declaration</strong>: Next, we create a generation_pipeline using the <code>pipeline</code> function from the Transformers library. This pipeline is configured to use the <code>declare-lab/flan-alpaca-base model</code>, which is a pre-trained model suitable for text generation.</li>\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:55]:
```suggestion
    "            <li><strong>Pipeline Declaration</strong>: Next, we create a generation_pipeline using the <code>pipeline</code> function from the Transformers library. This pipeline is configured to use the <code>declare-lab/flan-alpaca-large model</code>, which is a pre-trained model suitable for text generation.</li>\n",
```

**Review by B-Step62 — APPROVED** [review]
LGTM!
QQ: Can we re-run notebooks and deploy the changes after MLflow 2.9.0 release? Not a blocker, but those annoying pydantic warnings should be gone in the new version.

File: docs/source/llms/transformers/tutorials/translation/component-translation.ipynb [hunk 17]
```diff
@@ -0,0 +1,790 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
```
Comment (B-Step62) [inline docs/source/llms/transformers/tutorials/translation/component-translation.ipynb:356]: nit: Can we remove this warning? It's shown many times across the notebooks.

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb (hunk 16 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:55]: ah good catch!

File: docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb [hunk 18]
```diff
@@ -0,0 +1,505 @@
...
+   ],
+   "source": [
+    "# Validate that our loaded pipeline, as a generic pyfunc, can produce an output that makes sense\n",
+    "predictions = sentence_generator.predict(\n",
+    "    data=[\n",
+    "        \"I can't decide whether to go hiking or kayaking this weekend. Can you help me decide?\",\n",
+    "        \"Please tell me a joke about hiking.\",\n",
+    "    ],\n",
+    "    params={\"temperature\": 0.7},\n",
+    ")"
```
[the remark this answers is not in the document]
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/text-generation/text-generation.ipynb:427]: great point. Removed that extra markdown cell and concatenated the code cells :)

File: docs/source/llms/transformers/tutorials/translation/component-translation.ipynb (hunk 17 above)
Comment (BenWilson2) [reply docs/source/llms/transformers/tutorials/translation/component-translation.ipynb:356]: Yep! I have added warnings silencing for all UserWarnings (pydantic, setuptools) and added the fix for the tokenizer parallelization warning.
